docs/src/development/vm_integration_tests.md

```
@@ -0,0 +1,371 @@
+  `check.yaml` workflow. A VM test failure does not block or alter `check.yaml` results.
+
+### Adding a New Test
+
+1. Decide on the test scope:
+   - **Single service, one host:** Add `server.tests.units.<name>` to the service's
+     host configuration file.
+   - **Multi-node or cross-service:** Create a directory `tests/scenarios/<name>/`
+     with a `test.nix` file.
+1. The test appears automatically in `nixosTestConfigurations` — no registration needed.
+1. Run locally to verify:
+   ```bash
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)